File: app/src/main/java/com/test/homeautomation/screens/main/recycler/RecyclerViewAdapter.java
package com.test.homeautomation.screens.main.recycler;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import com.test.homeautomation.R;
import com.test.homeautomation.models.Device;

import java.util.ArrayList;
import java.util.List;

public class RecyclerViewAdapter extends RecyclerView.Adapter<ViewHolder> {

    private ArrayList<Device> devices;

    public RecyclerViewAdapter(ArrayList<Device> devices) {
        this.devices = devices;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.device_status_row, viewGroup, false);
        ViewHolder holder = new ViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, int i) {
        viewHolder.bind(devices.get(i));
    }

    @Override
    public int getItemCount() {
        return devices.size();
    }

    public void setDevices(List<Device> newDevicesList) {
        this.devices.clear();
        this.devices.addAll(newDevicesList);
        this.notifyDataSetChanged();
    }

}

File: app/src/main/java/com/test/homeautomation/screens/main/recycler/RecyclerViewAdapter.kt
package com.test.homeautomation.screens.main.recycler

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.test.homeautomation.R
import com.test.homeautomation.models.Device
import java.util.*

class RecyclerViewAdapter(private val devices: ArrayList<Device>) : RecyclerView.Adapter<ViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val view = LayoutInflater.from(viewGroup.context)
            .inflate(
                R.layout.device_status_row,
                viewGroup,
                false
            )
        return ViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) = viewHolder.bind(devices[i])

    override fun getItemCount(): Int = devices.size

    fun setDevices(newDevicesList: List<Device>) {
        this.devices.clear()
        this.devices.addAll(newDevicesList)
        this.notifyDataSetChanged()
    }

}

